### PR TITLE
Add requestedConfsDelay optional field vrfv2

### PR DESCRIPTION
**Pull request by makramkd**
Ticket: https://app.shortcut.com/chainlinklabs/story/21458/add-requestedconfsdelay-to-avoid-backup-and-primary-submission-collision

Add the requestedConfsDelay optional field for vrfv2 jobs, which
specifies the # of additional blocks to wait in addition to the
user-provided minimum request confirmations before servicing a VRF v2
request.

This is so that we avoid proof submission collisions between the primary
and backup VRF nodes in certain cases, which would cause transactions to
revert and funds to be unnecessarily spent.

### CODE REVIEW

File: core/testdata/testspecs/v2_specs.go [hunk 1]
```diff
@@ -218,6 +220,10 @@ func GenerateVRFSpec(params VRFSpecParams) VRFSpec {
 	if params.MinIncomingConfirmations != 0 {
 		confirmations = params.MinIncomingConfirmations
 	}
+	requestedConfsDelay := 0
+	if params.RequestedConfsDelay != 0 {
+		requestedConfsDelay = params.RequestedConfsDelay
+	}
```
Comment (PiotrTrzpil) [inline core/testdata/testspecs/v2_specs.go:226]: this could just be `requestedConfsDelay := params.RequestedConfsDelay` without an `if`

**Comment by PiotrTrzpil** [comment]
@makramkd looks good, though it likely also needs a CHANGELOG update explaining it.

**Comment by makramkd** [comment]
Excellent point @PiotrTrzpil , was thinking the same but it slipped my mind!

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 636d11c83964058cd299372f2f6f39af17bcdd86) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Comment by jkongie** [comment]
Let's add this field to the GQL API as well so that we can construct the definition in the UI. Operator UI will be updated to convert use the GQL API for jobs in the next few days

**Review by PiotrTrzpil — APPROVED** [review]

**Pull request merged**